### PR TITLE
ft(adminpage): Admin can disable an account [#187364817]

### PR DESCRIPTION
**Pull request by Oliviier-dev**
## ℹ️ Description

As an admin: 

- Due to varying reasons like fraud, abuse, etc,... admin should be able to disable any user account.
- Admin dashboard should list all users with an option of disabling their accounts.

## 🧪 How can this be tested?

This can be tested on the admin dashboard under the admin/users route

## 📸 Screenshots
Screenshot 2024-07-08 at 15 44 51

## 📍 Type of change

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or Enhancement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## ✅ Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have created a loom video for the new feature or enhancement
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by tuyishimehope — DISMISSED** [review]
LGTM

File: src/components/admincomponents/TableComponents/ActionButton.tsx [hunk 1]
```diff
@@ -75,9 +74,31 @@ const ActionButton = ({ activeRole, refreshUsers, id }: ActionButtonProps) => {
...
+    if (token && id) {
+      try {
+        const response = await disableAccount({ id, token }).unwrap();
+        if (response.message === 'User account disabled successfully') {
+          toast.success(response.message);
+          refreshUsers();
+        } else {
+          toast.error(response.message);
+        }
+      } catch (error: any) {
```
Comment (shebz2023) [inline src/components/admincomponents/TableComponents/ActionButton.tsx]: catch (error: errorResponse) {

File: src/components/admincomponents/TableComponents/ActionButton.tsx [hunk 2]
```diff
@@ -2,9 +2,9 @@ import { useState, useEffect, useRef } from 'react';
 import { BiDotsHorizontalRounded } from 'react-icons/bi';
 import RoleModal from './RoleModal';
 import { IUser } from '../../../types';
-import { useAssignRoleMutation } from '../../../service/authApi';
+import { useAssignRoleMutation, useDisableAccountMutation } from '../../../service/authApi';
 import { ColorRing } from 'react-loader-spinner';
-import { ToastContainer, toast } from 'react-toastify';
+import { toast } from 'react-toastify';
 import { Button } from '../../rootcomponents/Button';
```
Comment (shebz2023) [inline src/components/admincomponents/TableComponents/ActionButton.tsx:12]: interface ErrorResponse {
  status: number;
  data: {
    message: string;
  };
}

**Review by shebz2023 — COMMENTED** [review]
Great Job @Oliviier-dev, some small changes highlighted

**Review by shebz2023 — APPROVED** [review]
LGTM

**Review by Ke-bean — APPROVED** [review]
LGTM

**Review by SergeInGithub — APPROVED** [review]
LGTM. Nice work! @Oliviier-dev

**Review by SergeInGithub — APPROVED** [review]
LGTM

**Pull request merged**